libs/common/Generator.cs

```diff
@@ -16,11 +16,22 @@ public static class Generator
         /// </summary>
         /// <param name="size">The length of the hex identifier string</param>
         /// <returns></returns>
-        public static string CreateHexId(int size = 40)
+        public static unsafe string CreateHexId(int size = 40)
         {
             Span<byte> nodeIdBuffer = stackalloc byte[size / 2];
             RandomNumberGenerator.Fill(nodeIdBuffer);
-            return Convert.ToHexString(nodeIdBuffer).ToLowerInvariant();
+
+            char* charBuffer = stackalloc char[nodeIdBuffer.Length * 2]; // not the same as size (if size is odd)
```

Comment:
use string.Create if possible? Then you have one stackalloc place least to care about.

Reply:
Honestly, I wonder if it might be a better idea to just nuke the parameter :)

Reply:
it seems safe to nuke the parameter and always return a string of size 40. @vazois?

Reply:
Yes, it should be fine. We use the method only to generate node-ids which are of fixed size 40 characters.

Reply:
This is on the public API, so it would be bad form to remove it; I've instead switched to "stackalloc for small, lease for large" strategy that should work for everyone

Reply:
@Tornhoof re `string.Create`: that's not ideal here, because you can't use a `Span<T>` as a generic type argument (`TState`); sure, we could switch to `string.Create` passing in a `(byte[], int)` tuple, but that seems to make things even messier; I've left it as 2 length-constrained `stackalloc` spans (with array-pool fallback) using `new string(ReadOnlySpan<char>)`

Reply:
Well, there's _a way_ to do that but it's not worth it here 😆 

https://github.com/dotnet/runtime/blob/07648642d345087d344bc8b7b9b3ea7c8793bbea/src/libraries/Common/src/System/HexConverter.cs#L196-L197